src/libmongoc/tests/unified/operation.c

```diff
@@ -2761,6 +2761,48 @@ operation_assert_number_connections_checked_out (test_t *test,
    return true;
 }
 
+static bool
+operation_rename (test_t *test,
+                  operation_t *op,
+                  result_t *result,
+                  bson_error_t *error)
+{
+   // First validate the arguments
+   const char *object = op->object;
+   bson_iter_t it;
+   bool okay = bson_iter_init_find (&it, op->arguments, "to");
```

Comment:
I suggest parsing the arguments with the `bson_parser_t`. Example:

```c
   bson_parser_t *bp;
   char* to = NULL;

   bp = bson_parser_new ();
   bson_parser_utf8 (bp, "to", &to);
   if (!bson_parser_parse (bp, op->arguments, error)) {
      goto done;
   }

   /* other code */
done:
   bson_parser_destroy_with_parsed_fields (bp);
```

It will error on extra unrecognized fields by default. That may make a clearer error if we sync tests that add arguments to the "rename" operation.